topics/ui/principles/empty_state.md

```diff
@@ -10,144 +10,261 @@
+### Explain what to do when there is no action`
+
+If there is no action that can be performed by clicking a link,
+add a text describing all the required steps:
+
+<img src="empty_state_no_actions.png" alt="Explain what to do when there is no action" width="706"/>
+
+### Hide the toolbar
+
+Hide the area's toolbar if it does not have the same action as
+in the empty state. Usually, all other toolbar actions are not
+relevant in this case.
+
+<format color="369650" style="bold">Correct</format>
+
+<img src="empty_state_no_actions.png" alt="Hide the toolbar for the empty state" width="706"/>
+
+<format color="E55765" style="bold">Incorrect</format>
+
+<img src="empty_state_toolbar.png" alt="Toolbar is not needed" width="706"/>
+
+## Help topic
+<table style="none" border="false" column-width="fixed">
+  <tr>
+    <td width="378">
+      <img src="empty_state_link_to_help.png" alt="Link to a help topic"/>
+    </td>
+    <td>
+    <control>Link to a help topic</control>
+    <p>In tool windows and <a anchor="empty-state-of-the-master-detail-layout">master-detail layouts</a>,
+provide a link to a help topic that explains the functionality.
+Add the question mark icon in the beginning.</p>
+    </td>
+  </tr>
+  <tr>
+    <td width="378">
+      <img src="empty_state_inline_help.png" alt="Inline hint under the table"/>
+    </td>
+    <td>
+      <control>Context help in inline hints</control>
+      <p>In tables, trees, and lists, provide instructions according to the <a href="context_help.md" anchor="inline-help-text">inline text help</a> rules.
+Smaller UI areas rarely require an in-depth explanation compared to complex tool windows.
+A short help text should be enough, and it does not require switching contexts.</p>
+</td>
+</tr>
+</table>
 
-![](empty_state_database-tw-segment.png){width=262}
+## Empty state of the master-detail layout
 
-**Tables, trees and lists:** provide instructions according to the [Context help](context_help.md) rules. Smaller UI areas rarely require an in-depth explanation compared to complex tool windows. A short help text should be enough, and it does not require switching contexts.
+Master-detail layouts include a list or a tree on the left (in the master part)
+and the details area on the right.
 
-<format color="Red" style="bold">Incorrect</format>
+- If the master area is empty, display the explanation, action, and link to the help article.
+- If the details area is empty, no explanations are needed. In this case, show the actions to fill in the area.
+  The default pattern for the action is "_Select_ [entity] _to configure_".
 
-![](todo-filters-incorrect.png){width=456 style=block}
-*Do not use a help topic link in a table and UI areas other than tool windows.*
+<img src="empty_state_master_detail_layout.png" alt="Empty master-detail layout" width="706"/>
 
-<format color="Green" style="bold">Correct</format>
 
```

Comment:
There was one more pic in the old article with the master part having a list https://plugins.jetbrains.com/docs/intellij/empty-state.html#master-detail-layout.

Possibly, add the 3rd bullet point with a guideline something like "When there's content in the master part, provide a specific empty state in the detail area"?